### PR TITLE
Add support for Glue metastore cache

### PR DESCRIPTION
**Pull request by gs-asrivastav**
Sorry my [previous pull request](https://github.com/prestosql/presto/pull/1571) got corrupted in rebasing. 

Issue #1487

### CODE REVIEW

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 1]
```diff
@@ -38,23 +40,31 @@ protected void setup(Binder binder)
 
         if (buildConfigObject(HiveConfig.class).getRecordingPath() != null) {
             binder.bind(HiveMetastore.class)
-                    .annotatedWith(ForRecordingHiveMetastore.class)
```
Comment (Praveen2112) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: same for this

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 2]
```diff
@@ -22,14 +22,16 @@
...
+import io.prestosql.plugin.hive.metastore.cache.ForCachingHiveMetastore;
 import io.prestosql.spi.procedure.Procedure;
 
 import static com.google.inject.multibindings.Multibinder.newSetBinder;
 import static io.airlift.configuration.ConfigBinder.configBinder;
 import static org.weakref.jmx.guice.ExportBinder.newExporter;
 
 public class GlueMetastoreModule
-        extends AbstractConfigurationAwareModule
+    extends AbstractConfigurationAwareModule
```
Comment (Praveen2112) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: You can revert this changes, it was as per the code-style.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 3]
```diff
@@ -38,23 +40,31 @@ protected void setup(Binder binder)
...
-                    .in(Scopes.SINGLETON);
+                .annotatedWith(ForRecordingHiveMetastore.class)
+                .to(GlueHiveMetastore.class)
+                .in(Scopes.SINGLETON);
             binder.bind(GlueHiveMetastore.class).in(Scopes.SINGLETON);
             newExporter(binder).export(GlueHiveMetastore.class).withGeneratedName();
 
-            binder.bind(HiveMetastore.class).to(RecordingHiveMetastore.class).in(Scopes.SINGLETON);
+            binder.bind(HiveMetastore.class)
+                .annotatedWith(ForCachingHiveMetastore.class)
```
Comment (Praveen2112) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: indent this line

**Comment by ebyhr** [comment]
It would be better to update commit title. We shouldn't end with period `.` and the current title mixes uppercases "Module", "Metastore" and "Caching". 
How about simply `Add support for Glue metastore cache`?

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 3 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: Done.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 2 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: Done

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 1 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: Done

**Comment by gs-asrivastav** [comment]
@ebyhr Title updated as suggested.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 4]
```diff
@@ -44,17 +46,25 @@ protected void setup(Binder binder)
...
         }
         else {
-            binder.bind(HiveMetastore.class).to(GlueHiveMetastore.class).in(Scopes.SINGLETON);
+            binder.bind(HiveMetastore.class)
+                    .annotatedWith(ForCachingHiveMetastore.class)
+                    .to(GlueHiveMetastore.class)
+                    .in(Scopes.SINGLETON);
             newExporter(binder).export(HiveMetastore.class)
-                    .as(generator -> generator.generatedNameOf(GlueHiveMetastore.class));
+                .as(generator -> generator.generatedNameOf(GlueHiveMetastore.class));
```
Comment (ebyhr) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: Revert this change

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java [hunk 5]
```diff
@@ -44,17 +46,25 @@ protected void setup(Binder binder)
...
         }
+
+        binder.install(new CachingHiveMetastoreModule());
     }
-}
+}
```
Comment (ebyhr) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: Revert this change

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 4 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: Done!

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java (hunk 5 above)
Comment (gs-asrivastav) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java]: Done!

**Comment by ebyhr** [comment]
@gs-asrivastav Sorry, I mean commit message (not PR tittle). Also, could you squash the commits without  rebasing with upstream? Because the 2nd and 3rd commit diff aren't important. Even if you do force-push, we can confirm the diff on GitHub.

**Comment by gs-asrivastav** [comment]
@ebyhr Done!

**Comment by ebyhr** [comment]
Could you update the commit message too? Basically, we don't use following period `.`.

**Comment by gs-asrivastav** [comment]
@ebyhr Sorry i'd missed that, done!
